ext/pdo_mysql/tests/gh13384.phpt

```diff
@@ -0,0 +1,37 @@
+--TEST--
+GH-13384 Fixed GH-13167 Fixed the behavior when an inappropriate value was passed to `bindValue`.
+--EXTENSIONS--
+pdo_mysql
+--SKIPIF--
+<?php
+require_once __DIR__ . '/inc/mysql_pdo_test.inc';
+MySQLPDOTest::skip();
+?>
+--FILE--
+<?php
+require_once __DIR__ . '/inc/mysql_pdo_test.inc';
+$db = MySQLPDOTest::factory();
+$db->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
+$db->setAttribute(PDO::ATTR_EMULATE_PREPARES, false);
+
+$db->exec('CREATE TABLE test_gh13384 (mode TINYINT)');
+
+$stmt = $db->prepare('INSERT INTO test_gh13384 (mode) VALUES (?)');
+$stmt->bindValue(1, new DateTime(), PDO::PARAM_INT);
+
+var_dump($stmt->execute());
+var_dump($db->errorInfo());
+var_dump($db->query('SELECT * FROM test_gh13384')->fetchAll());
+?>
+--CLEAN--
+<?php
+require_once __DIR__ . '/inc/mysql_pdo_test.inc';
+$db = MySQLPDOTest::factory();
+$db->exec('DROP TABLE IF EXISTS test_gh13384');
+?>
+--EXPECTF--
+Fatal error: Uncaught Error: Object of class DateTime could not be converted to string in %s
```

Comment:
Is this correct? You are binding the param as an int, so why is this trying to convert it to a string?